src/main/scala/no/ndla/learningpathapi/repository/LearningPathRepositoryComponent.scala

```diff
@@ -287,55 +258,55 @@ trait LearningPathRepositoryComponent extends LazyLogging {
     }
 
     private def learningPathsWhere(whereClause: SQLSyntax)(
-        implicit session: DBSession = ReadOnlyAutoSession)
-      : List[LearningPath] = {
+        implicit session: DBSession = ReadOnlyAutoSession): List[LearningPath] = {
       val (lp, ls) = (LearningPath.syntax("lp"), LearningStep.syntax("ls"))
       sql"select ${lp.result.*}, ${ls.result.*} from ${LearningPath.as(lp)} left join ${LearningStep
         .as(ls)} on ${lp.id} = ${ls.learningPathId} where $whereClause"
         .one(LearningPath(lp.resultName))
         .toMany(LearningStep.opt(ls.resultName))
         .map { (learningpath, learningsteps) =>
-          learningpath.copy(
-            learningsteps = learningsteps.filter(_.status == StepStatus.ACTIVE))
+          learningpath.copy(learningsteps = learningsteps.filter(_.status == StepStatus.ACTIVE))
         }
         .list
         .apply()
     }
 
     private def learningPathWhere(whereClause: SQLSyntax)(
-        implicit session: DBSession = ReadOnlyAutoSession)
-      : Option[LearningPath] = {
+        implicit session: DBSession = ReadOnlyAutoSession): Option[LearningPath] = {
       val (lp, ls) = (LearningPath.syntax("lp"), LearningStep.syntax("ls"))
       sql"select ${lp.result.*}, ${ls.result.*} from ${LearningPath.as(lp)} left join ${LearningStep
         .as(ls)} on ${lp.id} = ${ls.learningPathId} where $whereClause"
         .one(LearningPath(lp.resultName))
         .toMany(LearningStep.opt(ls.resultName))
         .map { (learningpath, learningsteps) =>
-          learningpath.copy(
-            learningsteps = learningsteps.filter(_.status == StepStatus.ACTIVE))
+          learningpath.copy(learningsteps = learningsteps.filter(_.status == StepStatus.ACTIVE))
         }
         .single
         .apply()
     }
 
     def getLearningPathByPage(pageSize: Int, offset: Int)(
-        implicit session: DBSession = ReadOnlyAutoSession)
-      : List[LearningPath] = {
+        implicit session: DBSession = ReadOnlyAutoSession): List[LearningPath] = {
       val (lp, ls) = (LearningPath.syntax("lp"), LearningStep.syntax("ls"))
-      sql"select ${lp.result.*}, ${ls.result.*} from ${LearningPath.as(lp)} left join ${LearningStep.as(
-        ls)} on ${lp.id} = ${ls.learningPathId} offset $offset limit $pageSize"
-        .one(LearningPath(lp.resultName))
+      val lps = SubQuery.syntax("lps").include(lp)
+      sql"""
+            select ${lps.resultAll}, ${ls.resultAll} from (select ${lp.resultAll}
+                                                           from ${LearningPath.as(lp)}
+                                                           where document#>>'{status}' = ${LearningPathStatus.PUBLISHED.toString}
```

Comment:
Her returneres kun published stier. Det var ingen slik begrensing tidligere. Det er meininga?

Reply:
Det er meninga, det er denne funksjonen search-api bruker for å hente læringsstier for indeksering.

Reply:
👍